docs/docs/anatomy.md

```diff
@@ -0,0 +1,254 @@
+  import AccessStrategy = DynamoDB.AccessStrategy
+  import Category = DynamoDB.DynamoStoreCategory
+  import Context = DynamoDB.DynamoStoreContext
+  type Config = { context: Context; cache: ICache }
+  export function createCached<E, S, C>(name: string, codec_: ICodec<E, string, C>, fold: (s: S, e: E[]) => S, initial: S, access: AccessStrategy<E, S>, { context, cache }: Config) {
+    const caching = CachingStrategy.Cache(cache)
+    const codec = Codec.compress(codec_)
+    return Category.create(context, name, codec, fold, initial, caching, access);
+  }
+  export function createUnoptimized<E, S, C>(name: string, codec: ICodec<E, string, C>, fold: (s: S, e: E[]) => S, initial: S, config: Config) {
+    const access = AccessStrategy.Unoptimized()
+    return Dynamo.createCached(name, codec, fold, initial, access, config)
+  }
+  export function createLatestKnown<E,S,C>(name: string, codec: ICodec<E, string, C>, fold: (s: S, e: E[]) => S, initial: S, config: Config) {
+    const access = AccessStrategy.LatestKnownEvent()
+    return Dynamo.createCached(name, codec, fold, initial, access, config)
+  }
+  export function createSnapshotted<E,S,C>(name: string, codec: ICodec<E, string, C>, fold: (s: S, e: E[]) => S, initial: S, isOrigin: (e: E) => boolean, toSnapshot: (s: S) => E, config: Config) {
+    const access = AccessStrategy.Snapshot(isOrigin, toSnapshot)
+    return Dynamo.createCached(name, codec, fold, initial, access, config)
+  }
+  export function createRollingState<E,S,C>(name: string, codec: ICodec<E, string, C>, fold: (s: S, e: E[]) => S, initial: S, toSnapshot: (s: S) => E, config: Config) {
+    const access = AccessStrategy.RollingState(toSnapshot)
+    return Dynamo.createCached(name, codec, fold, initial, access, config)
+  }
+}
+
+// prettier-ignore
+export namespace MemoryStore {
+  export function create<E, S, C>(name: string, codec: ICodec<E, string, C>, fold: (s: S, e: E[]) => S, initial: S, { context: store }: { context: VolatileStore<string> }) {
+    return MemoryStoreCategory.create(store, name, codec, fold, initial)
+  }
+}
+```
+
+</details>
+
+```ts
+import * as Config from "../equinox-config"
+class Service {
+  // ... as above
+
+  // prettier-ignore
+  static resolveCategory(config: Config.Config) {
+    switch (config.store) {
+      case Config.Store.Memory:
+        return Config.MemoryStore.create(Stream.name, Events.codec, Fold.fold, Fold.initial, config)
+      case Config.Store.MessageDB:
+        return Config.MessageDB.createUnoptimized(Stream.name, Events.codec, Fold.fold, Fold.initial, config)
+      case Config.Store.Dynamo:
+        return Config.Dynamo.createUnoptimized(Stream.name, Events.codec, Fold.fold, Fold.initial, config)
+    }
+  }
+  static create(config: Config.Config) {
+    const category = Service.resolveCategory(config)
+    const resolve = (appointmentId: AppointmentId, userId: UserId) =>
+      Decider.forStream(category, Stream.streamId(appointmentId, userId), null)
```

Comment:
name the null and/or leave it optional?

Reply:
ATM the null is necessary, will be making it optional soon 👍

Reply:
see #41